lib/pages/signup.dart

```diff
@@ -0,0 +1,583 @@
+          child: Container(
+            alignment: Alignment.centerLeft,
+            decoration: kBoxDecorationStyle,
+            height: 50.0,
+            child: TextFormField(
+              textAlign: TextAlign.center,
+              controller: _password,
+              validator: (value) {
+                if (value.isEmpty) {
+                  return "Please confirm your password";
+                }
+                return null;
+              },
+              obscureText: true,
+              style: TextStyle(
+                color: Colors.white,
+                fontFamily: 'OpenSans',
+              ),
+              decoration: InputDecoration(
+                border: InputBorder.none,
+                contentPadding: EdgeInsets.only(top: 14.0),
+                prefixIcon: Icon(
+                  Icons.lock,
+                  color: Colors.white,
+                ),
+                hintText: 'Enter your Password',
+                hintStyle: kHintTextStyle,
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildConfirmPasswordTF() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        Text(
+          'Confirm Password',
+          style: kLabelStyle,
+        ),
+        SizedBox(height: 10.0),
+        Opacity(
+          opacity: 0.8,
+          child: Container(
+            alignment: Alignment.centerLeft,
+            decoration: kBoxDecorationStyle,
+            height: 50.0,
+            child: TextFormField(
+              textAlign: TextAlign.center,
+              controller: _confirmPass,
+              validator: (value) {
+                if (value.isEmpty) {
+                  return "Please confirm your password";
+                }
+                if (value != _password.text)
+                  return 'Password and confirmation password do not match.';
+                return null;
+              },
+              obscureText: true,
+              style: TextStyle(
+                color: Colors.white,
+                fontFamily: 'OpenSans',
+              ),
+              decoration: InputDecoration(
+                border: InputBorder.none,
+                contentPadding: EdgeInsets.only(top: 14.0),
+                prefixIcon: Icon(
+                  Icons.lock,
+                  color: Colors.white,
+                ),
+                hintText: 'Confirm your Password',
+                hintStyle: kHintTextStyle,
+              ),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildRegisterBtn() {
+    return Container(
+      padding: EdgeInsets.symmetric(vertical: 25.0),
+      width: double.infinity,
+      child: RaisedButton(
+        elevation: 5.0,
+        padding: EdgeInsets.all(15.0),
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(30.0),
+        ),
+        color: Colors.white,
+        onPressed: () async {
```

Comment:
Extract this to it's own function `registerNewUser()`

Reply:
Done. But please check if it is done properly.

Reply:
Done.